src/Components/Components/src/CascadingValueSource.cs

```diff
@@ -96,7 +96,8 @@ public Task NotifyChangedAsync()
             {
                 tasks.Add(dispatcher.InvokeAsync(() =>
                 {
-                    foreach (var subscriber in subscribers)
+                    Span<ComponentState> subscribersCopy = [.. subscribers];
```

Comment:
Does the use of `Span` here help in some way? I just did a small check with decompiling a console app and it looks like this is equivalent to:

```cs
Span<ComponentState> subscribersCopy = new Span<ComponentState>(list.ToArray());
```

In other words the code has the same allocation behavior as:

```cs
var subscribersCopy = list.ToArray();
```

I'm not aware of a nonallocating way to copy the list (since it's a reference type, `stackalloc` isn't on offer) besides using an array pool. Unless C# has added something new here that I missed. While array pooling would avoid the allocation, it's not necessarily a net gain in perf since it introduces cross-thread contention, and a single gen0 allocation is pretty low impact anyway.

So altogether, if you did the `Span` and `[.. subscribers]` as a matter of stylistic preference over `.ToArray()` that's fine. I'm only raising this point because many other devs might *think* modern C# is doing something magic with `Span` and `[.. subscribers]` but it appears not, and `.ToArray()` could avoid any such confusion.

Reply:
`ToArray()` sounds better here, IMO. More straightforward.

Reply:
Thanks for looking into that @SteveSandersonMS - I was going to do some analysis on that this morning (didn't get to it yesterday), but it looks like you beat me to it 🙂 

I'll switch to `.ToArray()`.

Reply:
Just did a quick check, and it looks like if the `Span<>` is known to be a fixed length then the compiler can store its elements on the stack (by generating a struct with `[InlineArray]`). This obviously doesn't apply to this case, but we can still manually create an `InlineArray` up to some fixed amount before falling back on `.ToArray()`.

I've updated the implementation to do this.